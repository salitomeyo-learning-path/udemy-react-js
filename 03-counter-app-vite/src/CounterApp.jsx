import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const handleAdd = ( event, newValue ) => {
        console.log('+1');
    }


    return (
        <>
            <h1>CounterApp</h1>
            <h2> { value } </h2>

            <button onClick={ (event) => handleAdd(event, 'Hello') }>+1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

