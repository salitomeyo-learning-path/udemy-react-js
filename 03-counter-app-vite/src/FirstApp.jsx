import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle }) => {

    // if ( !title ) {
    //     throw new Error('I need a title')
    // }

    return (
        <>
            <h1>{ title }</h1>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}

