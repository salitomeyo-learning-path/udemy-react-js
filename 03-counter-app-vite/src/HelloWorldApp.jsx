// function App() {
//     return <h1>Hello world</h1>;
// }

// export default App;

export const HelloWorldApp = () => {
    return <h1>Hello world</h1>;
}
