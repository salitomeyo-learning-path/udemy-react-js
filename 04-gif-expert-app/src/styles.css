* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: #323232;
    color: #eaeaea;
  }
  
  body {
    padding: 60px;
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
  }
  
  button {
    background-color: #12A5BC;
    font-size: 1rem;
    padding: 10px 30px;
    border-radius: 5px;
    border: none;
  }

  button:hover {
    background-color: #0f8da1;
  }

  .flex-box {
    display: flex;
  }

  .center {
    text-align: center;
  }

  .item {
    flex-grow: 1; /* default 0 */
  }

  .item-4 {
    flex-grow: 12; /* default 0 */
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }