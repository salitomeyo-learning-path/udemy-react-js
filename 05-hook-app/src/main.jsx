import React from 'react'
import ReactDOM from 'react-dom/client'
import { Memorize } from './06-memos/Memorize'

import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Memorize />
  </React.StrictMode>
)
