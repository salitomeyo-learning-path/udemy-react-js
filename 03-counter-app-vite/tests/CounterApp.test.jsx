import { render, screen, fireEvent } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';
import React from "react";


describe('Testing <CounterApp> Component', () => {
    const initialValue = 100
    
    test('debe de hacer match con el snapshot', () => {
        const { container } = render( 
        <CounterApp
            value={initialValue}
        /> );

        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar el valor inicial de 100', () => {
        render( 
        <CounterApp
            value={initialValue}
        /> );

        expect(screen.getByText(initialValue) ).toBeTruthy();
    })

    test('debe de incrementar con el boton +1', () => {
        render( 
        <CounterApp
            value={initialValue}
        /> );

        fireEvent.click( screen.getByText( '+1' ));

        expect( screen.getByText(initialValue+1) ).toBeTruthy();
    })

    test('debe de decrementar con el boton -1', () => {
        render( 
        <CounterApp
            value={initialValue}
        /> );

        fireEvent.click( screen.getByText( '-1' ));

        expect( screen.getByText(initialValue-1) ).toBeTruthy();
    })

    test('debe de decrementar con el boton -1', () => {
        render( 
        <CounterApp
            value={initialValue}
        /> );

        fireEvent.click( screen.getByText( '+1' ));
        fireEvent.click( screen.getByText( '+1' ));
        fireEvent.click( screen.getByText( '+1' ));
        fireEvent.click( screen.getByText( '+1' ));
        fireEvent.click( screen.getByRole('button', { name: 'reset-btn' }) )

        expect( screen.getByText(initialValue) ).toBeTruthy();
    })
})