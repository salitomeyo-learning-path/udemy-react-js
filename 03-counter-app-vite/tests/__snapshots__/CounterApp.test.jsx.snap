// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CounterApp> Component debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="reset-btn"
  >
    reset
  </button>
</div>
`;
